src/main/resources/assets/admin/common/js/ui/dialog/multistep/MultiStepDialog.ts

```diff
@@ -0,0 +1,281 @@
+import {DialogStep} from './DialogStep';
+import {Element} from '../../../dom/Element';
+import * as Q from 'q';
+import {ModalDialog, ModalDialogConfig} from '../ModalDialog';
+import {ActionButton} from '../../button/ActionButton';
+import {DivEl} from '../../../dom/DivEl';
+import {i18n} from '../../../util/Messages';
+import {Action} from '../../Action';
+import {AppHelper} from '../../../util/AppHelper';
+import {DefaultErrorHandler} from '../../../DefaultErrorHandler';
+
+
+export interface MultiStepDialogConfig
+    extends ModalDialogConfig {
+
+    steps: DialogStep[];
+
+    submitCallback?: { (): void; };
+
+    closeOnSubmit?: boolean;
+}
+
+export class MultiStepDialog
+    extends ModalDialog {
+
+    protected steps: DialogStep[];
+
+    protected config: MultiStepDialogConfig;
+
+    protected currentStep: DialogStep;
+
+    private stepsContainer: Element;
+
+    private forwardButton: ActionButton;
+
+    private backButton: ActionButton;
+
+    private noStepsBlock?: Element;
+
+    private stepCounter: Element;
+
+    private currentStepDataChangedHandler: { (): void };
+
+    protected constructor(config: MultiStepDialogConfig) {
+        super(config);
+    }
+
+    protected initElements(): void {
+        super.initElements();
+
+        this.steps = this.config.steps;
+        this.stepsContainer = this.createStepsContainer();
+        this.stepCounter = new DivEl();
+        this.backButton = this.addAction(new Action(i18n('dialog.multistep.previous')));
+        this.getButtonRow().addElement(this.stepCounter);
+        this.forwardButton = this.addAction(new Action(i18n('action.next')));
+    }
+
+    protected createStepsContainer(): Element {
+        return new DivEl();
+    }
+
+    protected initListeners(): void {
+        super.initListeners();
+
+        this.whenShown(() => {
+            this.handleShown();
+        });
+
+        this.onHidden(() => {
+            this.handleHidden();
+        });
+
+        this.forwardButton.getAction().onExecuted(() => {
+            if (this.forwardButton.isEnabled()) {
+                this.forwardOrSubmit();
+            }
+        });
+
+        this.backButton.getAction().onExecuted(() => {
+            this.showPreviousStep();
+        });
+
+        this.currentStepDataChangedHandler = AppHelper.debounce(this.handleCurrentStepDataChanged.bind(this), 300);
+    }
+
+    protected forwardOrSubmit(): void {
+        if (this.isLastStep()) {
+            this.submit();
+        } else {
+            this.showNextStep();
+        }
+    }
+
+    protected isLastStep(): boolean {
+        return this.currentStep && this.currentStep === this.steps.slice().pop();
+    }
+
+    protected isFirstStep(): boolean {
+        return this.currentStep === this.steps[0];
+    }
+
+    protected submit(): void {
+        if (this.config.submitCallback) {
+            this.config.submitCallback;
```

Comment:
Should be `this.config.submitCallback()`

Reply:
Updated